### PR TITLE
SCRUM-3712 Load AGM Phenotype Annotation data from FMS

### PR DESCRIPTION
**Pull request by markquintontulloch**

### CODE REVIEW

**Comment by oblodgett** [comment]
@adamgibs can you look at the UI side of this?

File: src/main/java/org/alliancegenome/curation_api/enums/FmsConditionRelation.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.alliancegenome.curation_api.enums;
+
+public enum FmsConditionRelation {
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/enums/FmsConditionRelation.java]: Maybe a better name would be ConditionRelationFmsEnum?

File: src/main/java/org/alliancegenome/curation_api/model/entities/PhenotypeAnnotation.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	@IndexedEmbedded(includePaths = {"name", "name_keyword"})
+	@IndexingDependency(reindexOnUpdate = ReindexOnUpdate.SHALLOW)
+	@ManyToOne
+	@JsonView({ View.FieldsOnly.class, View.ForPublic.class })
+	private VocabularyTerm relation;
+	
+	@Transient
+	public abstract String getSubjectCurie();
+
+	@Transient
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/model/entities/PhenotypeAnnotation.java]: I don't think these need to be @Transient given that they are abstract.

**Comment by adamgibs** [comment]
@markquintontulloch UI part looks good.

File: src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+CREATE TABLE agmphenotypeannotation_gene (
+	agmphenotypeannotation_id bigint,
+	assertedgenes_id bigint
+);
+
+ALTER TABLE agmphenotypeannotation_gene ADD CONSTRAINT agmphenotypeannotation_gene_agmphenotypeannotation_id_fk FOREIGN KEY (agmphenotypeannotation_id) REFERENCES agmphenotypeannotation (id);
+ALTER TABLE agmphenotypeannotation_gene ADD CONSTRAINT agmphenotypeannotation_gene_assertedgenes_id_fk FOREIGN KEY (assertedgenes_id) REFERENCES gene (id);
+CREATE INDEX agmphenotypeannotation_gene_agmphenotypeannotation_index ON agmphenotypeannotation_gene USING btree (agmphenotypeannotation_id);
+
```
Comment (oblodgett) [inline src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql:50]: I think there is a missing index on assertedgenes_id?

File: src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+CREATE TABLE allelephenotypeannotation_gene (
+	allelephenotypeannotation_id bigint,
+	assertedgenes_id bigint
+);
+
+ALTER TABLE allelephenotypeannotation_gene ADD CONSTRAINT allelephenotypeannotation_gene_gene_allelephenotypeannotation_id_fk FOREIGN KEY (allelephenotypeannotation_id) REFERENCES allelephenotypeannotation (id);
+ALTER TABLE allelephenotypeannotation_gene ADD CONSTRAINT allelephenotypeannotation_gene_assertedgenes_id_fk FOREIGN KEY (assertedgenes_id) REFERENCES gene (id);
+CREATE INDEX allelephenotypeannotation_gene_allelephenotypeannotation_index ON allelephenotypeannotation_gene USING btree (allelephenotypeannotation_id);
+
```
Comment (oblodgett) [inline src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql:72]: I think there is a missing index on assertedgenes_id?

File: src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+INSERT INTO bulkload (id, backendbulkloadtype, name, bulkloadstatus, group_id)
+	SELECT nextval('bulkload_seq'), 'PHENOTYPE', 'WB Phenotype Load', 'STOPPED', id FROM bulkloadgroup WHERE name = 'Phenotype Bulk Loads';
+INSERT INTO bulkload (id, backendbulkloadtype, name, bulkloadstatus, group_id)
+	SELECT nextval('bulkload_seq'), 'PHENOTYPE', 'XBXL Phenotype Load', 'STOPPED', id FROM bulkloadgroup WHERE name = 'Phenotype Bulk Loads';
+INSERT INTO bulkload (id, backendbulkloadtype, name, bulkloadstatus, group_id)
+	SELECT nextval('bulkload_seq'), 'PHENOTYPE', 'XBXT Phenotype Load', 'STOPPED', id FROM bulkloadgroup WHERE name = 'Phenotype Bulk Loads';
+INSERT INTO bulkload (id, backendbulkloadtype, name, bulkloadstatus, group_id)
+	SELECT nextval('bulkload_seq'), 'PHENOTYPE', 'ZFIN Phenotype Load', 'STOPPED', id FROM bulkloadgroup WHERE name = 'Phenotype Bulk Loads';
+INSERT INTO bulkscheduledload (id, cronschedule, scheduleactive)
+	SELECT id, '0 0 22 ? * SUN-THU', true FROM bulkload WHERE backendbulkloadtype = 'PHENOTYPE';
```
Comment (oblodgett) [inline src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql]: Right now the scheduleactive is set to false on the scheduled loads... will create a migration in the future to set them to true.

**Review by oblodgett — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql (hunk 3 above)
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql:50]: Good spot - now added

File: src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql (hunk 4 above)
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql:72]: Added now

File: src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql (hunk 5 above)
Comment (markquintontulloch) [reply src/main/resources/db/migration/v0.31.0.4__agm_phenotype_annotation.sql]: Have set to false

File: src/main/java/org/alliancegenome/curation_api/enums/FmsConditionRelation.java (hunk 1 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/enums/FmsConditionRelation.java]: Done

File: src/main/java/org/alliancegenome/curation_api/model/entities/PhenotypeAnnotation.java (hunk 2 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/model/entities/PhenotypeAnnotation.java]: Removed

**Review by oblodgett — APPROVED** [review]

**Pull request merged**